mongo/change_stream.go

```diff
@@ -256,8 +256,8 @@ func (cs *ChangeStream) executeOperation(ctx context.Context, resuming bool) err
 	if resuming {
 		cs.replaceOptions(cs.wireVersion)
 
-		csOptDoc := cs.createPipelineOptionsDoc()
-		if cs.err != nil {
+		csOptDoc, err := cs.createPipelineOptionsDoc()
+		if err != nil {
 			return cs.Err()
```

Comment:
It's confusing to check an error and then return a different error. Update this to return `err` (instead of `cs.Err()`) if it's not nil.